readthedocs/doc_builder/python_environments.py

```diff
@@ -72,6 +74,121 @@ def delete_existing_venv_dir(self):
             )
             shutil.rmtree(venv_dir)
 
+    def install_build_tools(self):
+        build_os = getattr(self.config.build, 'os', None)
+        build_tools = getattr(self.config.build, 'tools', None)
```

Comment:
these attributes should be always present now that you are checking if the config is using build_tools